pkg/operator/target_status.go

```diff
@@ -309,23 +309,28 @@ func patchPodMonitoringStatus(ctx context.Context, kubeClient client.Client, obj
 	return nil
 }
 
-// populateTargets populates the status object of each pod using the given
+// updateTargetStatus populates the status object of each pod using the given
 // Prometheus targets.
-func populateTargets(ctx context.Context, logger logr.Logger, kubeClient client.Client, targets []*prometheusv1.TargetsResult) error {
+func updateTargetStatus(ctx context.Context, logger logr.Logger, kubeClient client.Client, targets []*prometheusv1.TargetsResult) error {
 	endpointMap, err := buildEndpointStatuses(targets)
 	if err != nil {
 		return err
 	}
 
 	for job, endpointStatuses := range endpointMap {
+		// Kubelet scraping is configured through hard-coding and not through
+		// a PodMonitoring. As there's no status to update, we skip.
+		if strings.HasPrefix(job, "kubelet") {
+			continue
+		}
 		podMonitoringStatusContainer, err := buildPodMonitoring(job)
 		if err != nil {
-			return err
+			return errors.Wrapf(err, "building podmonitoring: %s", job)
 		}
 		podMonitoringStatusContainer.GetStatus().EndpointStatuses = endpointStatuses
 
 		if err := patchPodMonitoringStatus(ctx, kubeClient, podMonitoringStatusContainer, *podMonitoringStatusContainer.GetStatus()); err != nil {
-			logger.Error(err, "error patching job", "job", job)
+			return errors.Wrapf(err, "patching job: %s", job)
```

Comment:
I think the reason I made this a log and not a hard error is in case the PodMonitoring is removed between when we do the poll and we patch the status. We expect the patch to fail but we don't mind. Do you know what the implications for this are?